kubespawner/spawner.py

```diff
@@ -1768,11 +1771,51 @@ def get_env(self):
         """
 
         env = super(KubeSpawner, self).get_env()
+
+        """Separate env. variables into two dicts
+        - Dict containing only "valueFrom" env. variables, these are passed as-is.
+        - replace existing env dict with only "value" env. varaibles, these are passed into secret.
+        """
+        prepared_env = []
+        # create a separate dict for all "valueFrom" environment variables
+        extra_env = {k: v for k, v in (env or {}).items() if type(v) == dict}
+        # Replace existing env dict without "valueFrom" env. variables and pass it to secret
+        env = {k: v for k, v in (env or {}).items() if type(v) != dict}
+        for k, v in (extra_env or {}).items():
+            if not "name" in v:
+                v["name"] = k
+            extra_env[k] = v
+            prepared_env.append(get_k8s_model(V1EnvVar, v))
+
         # deprecate image
         env['JUPYTER_IMAGE_SPEC'] = self.image
         env['JUPYTER_IMAGE'] = self.image
-
-        return env
+        
+        if self.internal_ssl:
+            """
+            cert_paths:
+                certificate path references
+            hub_ca:
+                Path to the hub certificate authority
+            """
+            with open(self.cert_paths['keyfile'], 'r') as file:
+                env['ssl.key'] = file.read()
+
+            with open(self.cert_paths['certfile'], 'r') as file:
+                env['ssl.crt'] = file.read()
+
+            with open(self.cert_paths['cafile'], 'r') as file:
+                env["notebooks-ca_trust.crt"] = file.read()
+
+            with open(self.internal_trust_bundles['hub-ca'], 'r') as file:
+                env["notebooks-ca_trust.crt"] = env[
+                    "notebooks-ca_trust.crt"
+                ] + file.read()
+            env['JUPYTERHUB_SSL_KEYFILE'] = self.secret_mount_path + "ssl.key"
+            env['JUPYTERHUB_SSL_CERTFILE'] = self.secret_mount_path + "ssl.crt"
+            env['JUPYTERHUB_SSL_CLIENT_CA'] = (self.secret_mount_path + "notebooks-ca_trust.crt")
+
+        return env, prepared_env
```

Comment:
> I consider introducing get_env_from alongside get_env here, I dislike the complexity we embed in this by returning a tuple.

Yes, this is a good point. I thought about this, creating a new `get_env_from` just for env. variables using "valueFrom" but in the moment i went with getting something up quicker rather than simpler :). You're right, i will take a look at this to make it simpler.